packages/components/src/edit-in-place-control/style.scss

```diff
@@ -0,0 +1,16 @@
+.components-edit-in-place-control__label {
+	padding: $grid-unit-30;
+	width: auto;
+}
+.components-edit-in-place-control__input {
+	margin-bottom: 1px;
+	text-indent: $grid-unit;
+	&.large {
+		width: 100%;
+		transition: width 0.3s;
+	}
+	&.small {
+		width: 50%;
+		transition: width 0.3s;
```

Comment:
The transition relies on an explicit width, which doesn't work so well when the length of the menu name forces the button to be a different size:
![longmenuname](https://user-images.githubusercontent.com/677833/107488491-64913f00-6bc2-11eb-8696-87cd97f1d531.gif)

Not sure what the best way to achieve the design would be.

Reply:
I'm not sure I understand. If the concern is about longer menu names, then we can truncate at a reasonable length:

<img width="438" alt="image" src="https://user-images.githubusercontent.com/191598/107571479-f2682c80-6bb8-11eb-90ee-eb8d184ed328.png">


Otherwise, I'd expect the transition to feel something like this:

![Menu Name Transition](https://user-images.githubusercontent.com/191598/107571403-dbc1d580-6bb8-11eb-8f7c-42632ca0d897.gif)